Prox/Prox/Data/EventsController.swift

```diff
@@ -47,25 +51,52 @@ class EventsProvider {
     }
 
     private func shouldShowEventForNotifications(event: Event, forLocation location: CLLocation) -> Bool {
-        return isEventToday(event: event) && isEventYetToHappen(event: event) && doesEvent(event: event, startAtTimeIntervalFromNow: eventStartNotificationInterval)
+        let now = Date()
+        return isValidFutureEvent(event: event, currentTime: now, forNotifications: true) || isValidCurrentEvent(event: event, currentTime: now, forNotifications: true)
     }
 
     private func shouldShowEventForPlaces(event: Event, forLocation location: CLLocation) -> Bool {
-        return isEventToday(event: event) && isEventYetToHappen(event: event) && doesEvent(event: event, startAtTimeIntervalFromNow: eventStartPlaceInterval)
+        let now = Date()
+        return isValidFutureEvent(event: event, currentTime: now) || isValidCurrentEvent(event: event, currentTime: now)
     }
 
-    private func doesEvent(event: Event, startAtTimeIntervalFromNow timeInterval: TimeInterval) -> Bool {
-        // event must start in 1 hour
-        let now = Date()
-        let maxStartTime = now.addingTimeInterval(timeInterval)
+    private func isValidFutureEvent(event: Event, currentTime: Date, forNotifications: Bool = false) -> Bool {
+        let startTimeInterval = forNotifications ? eventStartNotificationInterval : eventStartPlaceInterval
+        return isEventToday(event: event) && isFutureEvent(event: event, currentTime: currentTime) && doesEvent(event: event, startAtTimeInterval: startTimeInterval, fromTime: currentTime)
+    }
+
+    private func isValidCurrentEvent(event: Event, currentTime: Date, forNotifications: Bool = false) -> Bool {
+        return isCurrentEvent(event: event, currentTime: currentTime) && doesEvent(event: event, lastLessThan: maxEventDuration)
+    }
+
+    private func doesEvent(event: Event, startAtTimeInterval timeInterval: TimeInterval, fromTime time: Date) -> Bool {
```

Comment:
nit: -> `doesEvent(..., startInTimeIntereval, ...)`

"At" indicates a single moment in time to me so `startAtTimeInterval` makes me think you're checking exact times.